### DISCUSSION TITLE
Bidirectional keyboard(+ mouse?) sharing

### DISCUSSION THREAD

**Discussion by ptrkrlsrd** [debauchee/barrier#1468] [Ideas]
Hey,

# Problem
I have a feature request which might be an uncommon use case, but want to throw some light on it anyways. I often work on both a desktop computer and a laptop at the same time. Sometimes I find myself placing my laptop in front of my desktop keyboard while sharing the mouse between both using Barrier. This works great until I want to type something on my desktop PC. Of course I could place my keyboard in a reachable distance, but I don't have space for it on my desk. 

# Solution
Therefore I was thinking it would be useful to be able to share my laptops keyboard with my desktop, while still using the mouse (and keyboard) connected to the desktop. The issue here is that my laptop is the client, which I assume would make the implementation of such a feature difficult as it requires it to also become the server. 

What do you guys think about the idea?

**Comment by Whiskas101** [comment]
This seems like a very niche but a very "great to have if you ever encounter it" feature
